File: chad/src/main/java/org/woahoverflow/chad/commands/info/Chad.java
package org.woahoverflow.chad.commands.info;

import org.woahoverflow.chad.framework.obj.Command;
import org.woahoverflow.chad.framework.handle.MessageHandler;
import sx.blah.discord.handle.impl.events.guild.channel.message.MessageReceivedEvent;
import sx.blah.discord.util.EmbedBuilder;

import java.lang.management.ManagementFactory;
import java.util.HashMap;
import java.util.List;

/**
 * @author sho
 * @since 0.6.3 B2
 */
public class Chad implements Command.Class {
    @Override
    public final Runnable run(MessageReceivedEvent e, List<String> args) {
        return () -> {
            // Creates an embed builder, and adds links etc to it.
            EmbedBuilder embedBuilder = new EmbedBuilder();
            embedBuilder.withTitle("Chad");
            embedBuilder.withDesc("by woahoveflow");
            embedBuilder.appendField("Uptime", ManagementFactory.getRuntimeMXBean().getUptime() / 60000 +" minutes", true);
            embedBuilder.appendField("Ping", e.getClient().getShards().get(0).getResponseTime()+"ms", true);
            embedBuilder.appendField("GitHub", "http://woahoverflow.org/github", false);
            embedBuilder.appendField("Website", "http://woahoverflow.org/chad", false);

            // Sends
            new MessageHandler(e.getChannel(), e.getAuthor()).sendEmbed(embedBuilder);
        };
    }

    @Override
    public final Runnable help(MessageReceivedEvent e) {
        HashMap<String, String> st = new HashMap<>();
        st.put("chad", "Gives information about the bot.");
        return Command.helpCommand(st, "Chad", e);
    }
}

File: chad/src/main/java/org/woahoverflow/chad/commands/info/Chad.kt
package org.woahoverflow.chad.commands.info

import org.woahoverflow.chad.core.ChadVar
import org.woahoverflow.chad.framework.handle.MessageHandler
import org.woahoverflow.chad.framework.obj.Command
import org.woahoverflow.chad.framework.util.Util
import sx.blah.discord.handle.impl.events.guild.channel.message.MessageEvent
import sx.blah.discord.util.EmbedBuilder
import java.lang.management.ManagementFactory
import java.util.*

/**
 * Information about Chad
 *
 * @author sho
 */
class Chad : Command.Class {
    override suspend fun run(e: MessageEvent, args: MutableList<String>) {
        MessageHandler(e.channel, e.author).credit(ChadVar.VERSION).sendEmbed {
            withDesc("**Chad** by woahoverflow\n\n" +
                    "**Uptime** ${Util.fancyDate(ManagementFactory.getRuntimeMXBean().uptime)}\n" +
                    "**Ping** `${e.client.shards[0].responseTime}` ms\n" +
                    "**GitHub** https://woahoverflow.org/github" +
                    "\n\nIf there's an issue, please join our Discord at https://woahoverflow.org/discord")
            withUrl("https://woahoverflow.org/chad")
        }
    }

    override suspend fun help(e: MessageEvent) {
        val st = HashMap<String, String>()
        st["chad"] = "Gives information about the bot."
        Command.helpCommand(st, "Chad", e)
    }
}